### ISSUE TITLE
Box2D FixtureDef restitutionThreshold Missing

### ISSUE THREAD

**Issue by Kureika** [libgdx/libgdx#7152]
#### Issue details

The FixtureDef class seems to be missing the restitutionThreshold variable somehow, others such as shape, density, friction and restitution are all there.

I checked the source at [FixtureDef.java](https://github.com/libgdx/gdx-box2d/blob/master/gdx-box2d-core/src/main/java/com/badlogic/gdx/physics/box2d/FixtureDef.java) and it is present there, so it is odd that it is not available.

#### Reproduction steps/code

Try to create a fixture definition, other variables are available and auto-complete:

```
        FixtureDef fixtureDef = new FixtureDef();
        fixtureDef.shape = boxShape;
        fixtureDef.density = density;
        fixtureDef.friction = friction;
        fixtureDef.restitution = restitution;
```

fixtureDef.restitutionThreshold does not appear to exist.

IntelliJ shows this as the decompiled class file:

```
package com.badlogic.gdx.physics.box2d;

public class FixtureDef {
    public Shape shape;
    public float friction = 0.2F;
    public float restitution = 0.0F;
    public float density = 0.0F;
    public boolean isSensor = false;
    public final Filter filter = new Filter();

    public FixtureDef() {
    }
}
```

#### Version of libGDX and/or relevant dependencies
Maven dependency:

```
        <dependency>
            <groupId>com.badlogicgames.gdx</groupId>
            <artifactId>gdx-box2d</artifactId>
            <version>1.11.0</version>
        </dependency>
```

#### Stacktrace
n/a

#### Please select the affected platforms
- [ ] Android
- [ ] iOS
- [ ] HTML/GWT
- [x] Windows
- [ ] Linux
- [ ] macOS

**Comment by tommyettinger** [comment]
I'm not sure if the gdx-box2d on Maven Central was compiled from the older Box2D in the libGDX repo (which is based on 2.3.1 according to https://github.com/libgdx/libgdx/blob/master/extensions/gdx-box2d/gdx-box2d/jni/Box2D/CHANGES#L2 , 9 years old) or the Box2D in the independent libgdx/gdx-box2d repo (which is based on 2.4.1, about 3 years old). The restitutionThreshold variable was added in 2.4.1: https://github.com/libgdx/gdx-box2d/blob/master/gdx-box2d-core/jni/src/box2d-2.4.1/CHANGELOG.md?plain=1#L8

**Comment by Kureika** [comment]
Ah, I did not even realize that there were 2 different Box2Ds here, then it makes sense why it did not match the repo I checked.

I guess if the Maven Central version still uses a 9 year old version, that there are no immediate plans to update that then.

I suppose I need to try building it locally then.

**Comment by tommyettinger** [comment]
Yeah, it looks like Box2D would need at least some native code built, and that's a hassle I haven't dealt with for libGDX. If you don't need the 2.4.1 features and 2.3.1 is suitable, great! Otherwise, people are looking into building a newer Box2D version, mainly for rope joint support, and I'm not sure how that's going. An issue is that the GWT version of Box2D relies on a pure-Java port, which hasn't been updated to 2.4.1 as I understand it. The author of Box2D is also working on a Box2D 3.0, and that might be released in a year or less (maybe).

**Comment by ice1000** [comment]
> Ah, well if there are on-going efforts to update Box2D, I suppose I will have something to look forward to.

See also #7144 and https://github.com/ice1000/gdx-box2d

**Comment by ice1000** [comment]
> I'm not sure if the gdx-box2d on Maven Central was compiled from the older Box2D in the libGDX repo (which is based on 2.3.1 according to https://github.com/libgdx/libgdx/blob/master/extensions/gdx-box2d/gdx-box2d/jni/Box2D/CHANGES#L2 , 9 years old) or the Box2D in the independent libgdx/gdx-box2d repo (which is based on 2.4.1, about 3 years old). The restitutionThreshold variable was added in 2.4.1: https://github.com/libgdx/gdx-box2d/blob/master/gdx-box2d-core/jni/src/box2d-2.4.1/CHANGELOG.md?plain=1#L8

Beware that libgdx/gdx-box2d is still not up-to-date, it's missing a couple of commits from box2d upstream as well as some commits from libgdx upstream. #7144 has these commits ported.
